### DISCUSSION TITLE
WebSocket forwarding works on local codespaces, but not on the web

### DISCUSSION THREAD

**Discussion by fonsp** [community/community#3908] [Codespaces]
If I host a web server with websockets (over the same port as regular HTTP requests), it does not work when using GitHub Codespaces on the web, i.e. through a URL like `https://fonsp-pluto-jl-89fv-1234.githubpreview.dev/`.

When using Codespaces locally (MacOS), with the GitHub Codespaces extension, it works!

According to my web server, only the `Connection: close` HTTP request arrives at the server, the `Connection: upgrade` request does not. The same web server ([Pluto.jl](https://github.com/fonsp/Pluto.jl)) works through other proxied setups, like heroku, [`jupyter-server-proxy`](https://github.com/jupyterhub/jupyter-server-proxy), [`ngrok`](https://ngrok.com/), and more.

**Comment by Zehir** [comment]
I can confirm this behavior on my project too.
In local I can connect to the websocket but not from the web.
https://github.com/Zehir/deconz-ruler/

**Comment by tristan-f-r** [comment]
Can confirm. Also, afaik gitpod has websocket forwarding so it should be possible for Github Codespaces as well.

**Comment by nicolapalavecino** [comment]
I'm trying to setup my vite built + express backend project running in two different ports and the websocket support is making my frontend HMR not to work. It would be awesome if web based codespaces could support binding websockets too!

**Comment by hsein-bitar** [comment]
Any updates on this issue? I wanted to use MinIO in a container inside the code-space, tried to allow http upgrades using NGINX, and I was not able to retrieve the objects. I was forced to use an older MinIO docker image that did not rely on web-sockets.

**Comment by Alek99** [comment]
Same I'm looking to set up a code spaces for a project but I need websockets to connect from the web

**Comment by BarriBarri20** [comment]
I think is a problem related to remote connection with ssh clients, as gitpod.io also have the same problem!

**Comment by artrixdotdev** [comment]
any updates? this still doesnt work

**Reply by BarriBarri20** [reply]
try to switch to HTTP instead of HTTPS, this a logical explanation of the problem : https://stackoverflow.com/questions/74258122/codespaces-and-https

**Reply by karletzel** [reply]
I did this and still ran into errors, see my note below re: 308 error from Twilio

**Comment by karletzel** [comment]
FWIW I'm trying to get Twilio to forward SMS messages to my python code running in codespaces and having similar issue, twilio will not connect to the websocket, I get this error message: 

HTTP bad host name
ERROR: 11210
ERROR
Description
The DNS entry for the URL's host cannot be resolved.

Possible Causes
URL references a locally defined hostname (check your /etc/hosts )
URL references an undefined hostname

My messaging is configured like this where 8765 is the port #
![image](https://github.com/community/community/assets/151890630/9723cb1e-2e31-4c64-bd2d-4db53d1c9f88)

I also tried with the remote port # (127.0.0.1:8765) and that didn't work either. 

I did make the port public. 

I'm a total newb coder so apologies if this is nonsense or really basic...

**Reply by karletzel** [reply]
Based on another thread I switched the protocol to http, and now I get a 308 error in Twilio: 
![image](https://github.com/community/community/assets/151890630/2d023efb-1b27-4b58-857e-5365ab97a4a7)

I also tried to get ngrok from philnash to work but that never went anywhere, I never got the prompt to enter my authentication token, it just requested the port # the said it could not set up the tunnel.  

At this point should I just bail on Codespaces and set up my system in AWS, which is where I'll have to go for production anyways?  I don't have a team to collaborate with I'm just a lone developer working on a hobby project ATM.

**Comment by dmiyamasu** [comment]
I've got this same problem.
FYI - this is the Twilio dev article that I am trying to get up and running in codespaces: https://www.twilio.com/en-us/blog/voice-ai-assistant-openai-realtime-api-node

It requires my endpoint to accept incoming websocket connections "wss://<codepaces public url>"

**Reply by hydroweaver** [reply]
Hey @dmiyamasu I was able to get it running on Codespaces. I face this issue while creating a local server on codepaces and then forwarding the address
